q2_moshpit/busco/tests/test_busco.py

```diff
@@ -0,0 +1,359 @@
+        self.assertSetEqual(set(observed), set(expected))
+
+    def test_collect_summaries_and_save(self):
+        """
+        Test for `_collect_summaries_and_save` function.
+        Uses data stored in ./data. Checks for data frame equality.
+        """
+        with tempfile.TemporaryDirectory() as tmp_path:
+            path_to_summaries = {}
+            common_path = os.path.join(os.path.dirname(__file__), "data")
+
+            for i in range(1, 4):
+                path_to_summaries[f"sample{i}"] = os.path.join(
+                    common_path,
+                    f"batch_summary_sample{i}.txt"
+                )
+
+            observed = _collect_summaries_and_save(
+                path_to_run_summaries=path_to_summaries,
+                all_summaries_path=os.path.join(tmp_path, "aggregated.csv"),
+            )
+
+            p = os.path.join(common_path, "all_batch_summaries.csv")
+            expected = pd.read_csv(p)
+            pd.set_option('display.max_columns', None)
+            self.assertTrue(observed.equals(expected))
+
+    # Test `_draw_busco_plots`
+    def draw_n_busco_plots(self, filename, delim):
+        """
+        Creates plot from a table containing information about
+        one or more samples. Checks for the existence of the output
+        plots.
+
+        Args:
+            filename (str): name of file in ./data to construct the images
+            delim (str): delimiter of `filename`
+        """
+        # Create an empty dictionary to store the DataFrames
+        path_to_run_summaries = {}
+
+        # Group the DataFrame by the 'sample_id' column
+        p = os.path.join(
+            os.path.dirname(__file__),
+            f"data/{filename}"
+        )
+        df = pd.read_csv(p, delimiter=delim)
+        print(df)
+        grouped = df.groupby("sample_id")
+
+        # Iterate through the groups and store each group as a
+        # DataFrame in the dictionary
+        # Creates output directory with path 'tmp'
+        with tempfile.TemporaryDirectory() as tmp_path:
+            for sample_id, group_df in grouped:
+                path_to_df = f"{tmp_path}/{sample_id}.csv"
+                group_df.to_csv(path_to_df, sep="\t", index=False)
+                path_to_run_summaries[sample_id] = path_to_df
+
+            # Draw plots
+            paths_to_plots = _draw_busco_plots(
+                path_to_run_summaries=path_to_run_summaries,
+                plots_dir=os.path.join(tmp_path, "plots"),
+            )
+
+            # Check if busco plots are in fact generated
+            for _, value in paths_to_plots.items():
+                self.assertTrue(os.path.exists(value))
+
+    def test_draw_busco_plots_multiple(self):
+        self.draw_n_busco_plots(
+            filename="all_batch_summaries.csv", delim=","
+        )
+
+    def test_draw_busco_plots_one(self):
+        self.draw_n_busco_plots(
+            filename="batch_summary_sample1.txt", delim="\t"
+        )
```

Comment:
### The functions purpose
The `_draw_busco_plots` function essentially creates plots so that in the next step they can be zipped and included in the output (so it can be downloaded when viewing the `output.qzv`). It is otherwise irrelevant to the final output since the plot that is generated in the HTML is rendered in a separate function. 

### Breaking down the function
For each sample that is processed the following occurs:

1. The data frame from which the plot is constructed is formatted to have the expected column names and information.
2. The plot object is created and formatted.
3. The plot is saved and the location is written in a dictionary which is the final output of the function. 

The second step from above is hard to test as it would involve comparing images. As of 1 and 3 these seem trivial to me and I don't see a strong need to test them. Similarly, I would oppose splitting this code into too many functions it is only used once and would make it less readable.